test/store_test.dart

```diff
@@ -65,58 +65,70 @@ void main() {
     });
 
     test('should trigger in response to an action', () {
-      Action _action = new Action();
-      store.triggerOnAction(_action);
-      store.listen((Store payload) => expectAsync((payload) {
-            expect(payload, equals(store));
-          }));
-      _action();
+      var c = new Completer();
+      var action = new Action();
+      store.triggerOnAction(action);
+      store.listen(expectAsync((payload) {
+        c.complete();
+        expect(payload, equals(store));
+      }));
+      action();
+      return c.future;
     });
 
     test(
         'should execute a given method and then trigger in response to an action',
         () {
-      Action _action = new Action();
-      bool methodCalled = false;
+      var c = new Completer();
+      var action = new Action();
+      var methodCalled = false;
       syncCallback(_) {
         methodCalled = true;
       }
-      store.triggerOnAction(_action, syncCallback);
-      store.listen((Store payload) => expectAsync((payload) {
-            expect(payload, equals(store));
-            expect(methodCalled, equals(true));
-          }));
-      _action();
+      store.triggerOnAction(action, syncCallback);
+      store.listen(expectAsync((payload) {
```

Comment:
The `Completers` you've added are not necessary. You are duplicating the behavior of `expectAsync()`. The docs for `expectAsync()` say:
```
/// The test framework will wait for the callback to run the [count] times
/// before it considers the current test to be complete.
```